hermes/probe/check_nil.go

```diff
@@ -0,0 +1,159 @@
+	hashTime := time.Now()
+	hash := sha1.New()
+	if _, err := io.Copy(hash, reader); err != nil {
+		target.LatencyMetrics.ProbeOplatency[m.ProbeOps[m.VERIFY_FILE]][m.ExitStatuses[m.FILE_CORRUPTED]].Metric("latency").AddFloat64(time.Now().Sub(hashTime).Seconds())
+		return nil, fmt.Errorf("checkNilFile.compute_checksum_failed: unable to compute checksum of file contents: %w", err)
+	}
+	if fmt.Sprintf("%x", hash) != nilFilename[len("Hermes_NIL_"):] {
+		target.LatencyMetrics.ProbeOpLatency[m.ProbeOps[m.VERIFY_FILE]][m.ExitStatuses[m.FILE_CORRUPTED]].Metric("latency").AddFloat64(time.Now().Sub(hashTime).Seconds())
+		return fmt.Errorf("checkNilFile.verify_checksum_failed: checksum of file did not match filename: %w", err)
+	}
+	hashTime = time.Now().Sub(hashTime)
+	reader.Close()
+
+	// TODO(evanSpendlove): Decode the contents of the NIL file.
+	journal := &sjpb.Journal{}
+	if err := proto.Unmarshal(nilContents, journal); err != nil {
+		target.LatencyMetrics.ProbeOpLatency[m.ProbeOps[m.READ_FILE]][m.ExitStatuses[m.FILE_CORRUPTED]].Metric("latency").AddFloat64(time.Now().Sub(start).Seconds())
+		return fmt.Errrof("checkNilFile.unmarshal_failed: unable to unmarshal nil file proto: %w", err)
+	}
+	target.LatencyMetrics.ProbeOpLatency[m.ProbeOps[m.READ_FILE]][m.ExitStatuses[m.SUCCESS]].Metric("latency").AddFloat64(time.Now().Sub(start).Seconds())
+	logger.Infof("checkNilFile: unmarshal complete for nil file proto.")
+
+	// TODO(evanSpendlove): Verify the contents against in memory and checksum.
+	start = time.Now() // Start for verify op
+
+	// TODO(evanSpendlove): Handle intent from NIL file.
+	// Read intent, if not nil, check intent validity and update nil file in memory and defer nil file write.
+	if journal.Intent != nil {
+		// Check validity
+		fileID := journal.Intent.Filename[len("Hermes_"):len("Hermes_ID")]
+		switch journal.Intent.FileOperation {
+		case sjpb.Intent_CREATE:
+			// Check if the file with this filename has already been created
+			// TODO: Should we check the file system for the file existing?
+			if target.Journal.Filenames[fileID] != Journal.Intent.Filename {
+				// TODO(evanSpendlove): Call createFile() passing the fileID
+				logger.Info("File create not complete yet")
+			} else {
+				journal.Intent = nil // Remove intent as operation performed in full.
+			}
+		case sjpb.Intent_DELETE:
+			if _, ok := target.Journal.Filenames[fileID]; !ok {
+				// TODO(evanSpendlove): Call  DeleteFile() passing the fileID
+				logger.Info("File delete not complete yet")
+			} else {
+				journal.Intent = nil // Remove intent as operation performed in full.
+			}
+		}
+	}
+
+	if !reflect.DeepEqual(journal, target.Journal) {
+		target.LatencyMetrics.ProbeOpLatency[m.ProbeOps[m.VERIFY_FILE]][m.ExitStatuses[m.FILE_CORRUPTED]].Metric("latency").AddFloat64(time.Now().Add(hashTime).Sub(start).Seconds())
+		return fmt.Errorf("checkNilFile.file_inconsistent: nil file contents do not match expectations: %w", err)
```

Comment:
Where is `err` coming from here?

Reply:
There shouldn't be an `err` here, I've removed it.